lib/roby/app/scripts/run.rb

```diff
@@ -30,6 +30,12 @@
     opt.on '--port=PORT', Integer, 'the interface port' do |port|
         app.shell_interface_port = port
     end
+    opt.on '--rest[=PORT]', Integer, 'enable the experimental REST API' do |port|
```

Comment:
Would you consider allowing a `String` here? That would allow one to use the REST api over a unix socket, which I think suits better as an IPC mechanism (no worries about port being in use).  [Thin provides this with zero implementation cost](https://github.com/macournoyer/thin/blob/master/lib/thin/server.rb#L17)

Reply:
Particularly useful for the vscode extension, no need for random ports..

Reply:
Yeah ... I considered it, the problem is that the Ruby REST client I'm using doesn't support Unix sockets, so I ruled it out. Given that we have a good use-case for this, I'll re-add it.

Reply:
> Ruby REST client I'm using doesn't support Unix sockets

curl does support unix sockets

Reply:
> curl does support unix sockets

As in "rest client that is being used by a ruby script". I don't plan to shell out to curl in each and every test ;-)

In-browser REST clients also don't support Unix, but if I need to do manual tests then I can restart to use TCP.